### PR TITLE
feat(tags): add tags collection and metric

### PR DESCRIPTION
**Pull request by gsanchezgavier**
* Add tags collection from vcenter using the rest api
* Removes vendor files

### CODE REVIEW

File: cmd/nri-vsphere/nri-vsphere.go [hunk 1]
```diff
@@ -45,12 +46,22 @@ func main() {
 		config.Logrus.WithError(err).Fatal("failed to create client")
 	}
 
+	config.VMWareClientRest, err = client.NewRest(config.VMWareClient, config.Args.User, config.Args.Pass)
```
Comment (paologallinaharbur) [inline cmd/nri-vsphere/nri-vsphere.go:49]: We create the rest client and the tags manager event if the tags are not collected, should add a check?
There is not a real drawback from creating the client and the manager, but it is not needed

File: internal/collect/tags.go [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+	if err != nil {
+		return fmt.Errorf("failed to collect tags:%v", err)
+	}
+
+	dc.AddTags(tagsByObject)
+
+	return nil
+}
+
+func getTags(ref []mo.Reference, tm *tags.Manager, tc load.TagCategories) (map[types.ManagedObjectReference][]load.Tag, error) {
```
Comment (paologallinaharbur) [inline internal/collect/tags.go:85]: can you add comments to this funcion?

File: internal/collect/tags.go [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+}
+
+func collectTags(config *load.Config, managedObjectsSlice interface{}, dc *load.Datacenter) error {
+	var ref []mo.Reference
+
+	if !(config.Args.EnableVsphereTags && config.IsVcenterAPIType) {
+		return nil
+	}
+
+	switch obs := managedObjectsSlice.(type) {
```
Comment (paologallinaharbur) [inline internal/collect/tags.go:38]: why do we need this swich case? 
Is it possible to have the type outside this set?

Comment (gsanchezgavier) [reply internal/collect/tags.go:38]: i done it this way to simply call this func from every collector passing the slice of collected objects. I didn't find a way to do this better without generics

Comment (paologallinaharbur) [reply internal/collect/tags.go:38]: mmm I was thinking to ask for a less generic interface, something like:
```
func collectTags(config *load.Config, obs []mo.Entity, dc *load.Datacenter) error {
	var ref []mo.Reference

	if !(config.Args.EnableVsphereTags && config.IsVcenterAPIType) {
		return nil
	}

	for _, o := range obs {
		ref = append(ref, o.Reference())
	}
```

The issue is that the function ask for a slice of interface and there is not a way to cast it without using reflection 😢 
> https://stackoverflow.com/questions/12753805/type-converting-slices-of-interfaces

**Pull request merged**